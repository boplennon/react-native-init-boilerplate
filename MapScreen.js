import * as React from 'react';
import { Text, View } from "react-native";
import styles from './styles';

export const MapScreen = () => (
    <View style={styles.container}>
        <Text>Map</Text>
    </View>);

export default MapScreen;