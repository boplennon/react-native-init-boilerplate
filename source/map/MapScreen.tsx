import * as React from 'react';

import _ from 'lodash';
import * as types from './types';
import MapService from './MapService';
import { Text } from 'react-native';
import styles, { TabIconSize } from '../styles';
import { NavigationBottomTabScreenOptions } from 'react-navigation';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
// @ts-ignore
import MapboxGL from '@mapbox/react-native-mapbox-gl';

// TODO: get from app secure config
const accessToken = '';

interface IState {
  region?: types.IRegion;
  /**
   * User's physical location
   */
  userLocation?: types.ILatLng;
  mapMargin: number;
}

interface IProps {
  initialRegion?: types.IRegion;
}

export class MapScreen extends React.Component<IProps, IState> {

  constructor(props: IProps) {
    super(props);
    this.state = {
      mapMargin: 1,
      region: props.initialRegion || {
        latitude: 59.330839,
        longitude: 18.062976,
        latitudeDelta: 0.0059397161733585335,
        longitudeDelta: 0.005845874547958374,

      },
    };

    this.setMap = this.setMap.bind(this);
    this.onMapReady = this.onMapReady.bind(this);
    this.onUserLocationChange = this.onUserLocationChange.bind(this);
  }
  static navigationOptions: NavigationBottomTabScreenOptions = {
    tabBarIcon: <FontAwesome name='map-o' size={TabIconSize} color='white' />,
    tabBarLabel: <Text style={styles.tabBarLabel}>Map</Text>,
  };

  componentWillMount() {
    MapboxGL.setAccessToken(accessToken);
  }

  onMapReady() {
    this.setState({ mapMargin: 0 });
  }

  setMap(ref: any) {
    let mapInstance: any = null;
    if (ref && ref.root) {
      mapInstance = ref.root;
    } else if (ref) {
      mapInstance = ref;
    }

    if (mapInstance) {
      MapService.Instance.setMap(mapInstance);
    }
  }

  onUserLocationChange(coordiante: types.ILocation) {
    if (
      _.isEqual(this.state.userLocation, coordiante) ||
      MapService.Instance.isOnUserLocation
    ) {
      return;
    }

    this.setState({ userLocation: coordiante });
  }

  render() {
    return (
        <MapboxGL.MapView
          showUserLocation={true}
          zoomLevel={16}
          userTrackingMode={MapboxGL.UserTrackingModes.Follow}
          styleURL={MapboxGL.StyleURL.Street}
          style={{flex: 1, padding: 0, margin: 0}}
          ref={(c: any) => (MapService.Instance.setMap(c))}
        />
    );
  }
}

export default MapScreen;
