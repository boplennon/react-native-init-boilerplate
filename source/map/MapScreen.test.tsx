import React from 'react';
import { shallow } from 'enzyme';

import MapScreen from './MapScreen';

function setup() {
    return shallow(
        // @ts-ignore - Enzyme TypeScript issues
        <MapScreen />
    );
}

describe('components', () => {
    describe('MapScreen', () => {
        it('renders correctly with defaults', () => {
            const wrapper = setup();
            expect(wrapper).toMatchSnapshot();
        });
    });
});