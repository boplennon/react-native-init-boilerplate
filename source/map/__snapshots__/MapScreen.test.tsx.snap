// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components MapScreen renders correctly with defaults 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MapScreen />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": null,
      "loadingEnabled": true,
      "loadingIndicatorColor": "#057dc2",
      "maxZoomLevel": 18,
      "onMapReady": [Function],
      "onRegionChangeComplete": [Function],
      "provider": undefined,
      "showsBuildings": true,
      "showsCompass": true,
      "showsMyLocationButton": false,
      "showsUserLocation": true,
      "style": Array [
        Object {
          "flex": 1,
          "justifyContent": "space-between",
        },
        Object {
          "marginBottom": 1,
        },
      ],
      "toolbarEnabled": true,
    },
    "ref": [Function],
    "rendered": null,
    "type": Object {
      "Callout": "View",
      "Circle": "View",
      "MapView": "View",
      "Marker": "View",
      "Overlay": "View",
      "Ploygon": "View",
      "Polyline": "View",
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": null,
        "loadingEnabled": true,
        "loadingIndicatorColor": "#057dc2",
        "maxZoomLevel": 18,
        "onMapReady": [Function],
        "onRegionChangeComplete": [Function],
        "provider": undefined,
        "showsBuildings": true,
        "showsCompass": true,
        "showsMyLocationButton": false,
        "showsUserLocation": true,
        "style": Array [
          Object {
            "flex": 1,
            "justifyContent": "space-between",
          },
          Object {
            "marginBottom": 1,
          },
        ],
        "toolbarEnabled": true,
      },
      "ref": [Function],
      "rendered": null,
      "type": Object {
        "Callout": "View",
        "Circle": "View",
        "MapView": "View",
        "Marker": "View",
        "Overlay": "View",
        "Ploygon": "View",
        "Polyline": "View",
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
